File: week-1/sec_largest_elem/sec_largest.py
def second_largest(arr):
    unique_elements = list(set(arr))
    if len(unique_elements) < 2:
        return "Second largest element not founded."
    
    unique_elements.sort(reverse=True)
    return f"Second largest element is :{unique_elements[1]}"


arr = list(map(int, input("Enter array element separated by space : ").split()))


result = second_largest(arr)
print(result)

File: week-1/sec_largest_elem/sec_largest.c
// print the second largest element in array 

#include<stdio.h>

int main() {

    int n,i,max,second;

    printf("Enter the number of elements : ");
    scanf("%d",&n);

    int arr[n];
    printf("Enter the elemets for the array : ");
    for(int i=0; i<n; i++) {
        scanf("%d",&arr[i]);
    }

    max = second = -21345678;

    for(int i=0; i<n; i++) {
        if (arr[i] > max) {
            second = max;
            max = arr[i];
        }

        else if(arr[i] > second && arr[i] != max) {
            second = arr[i];
        }
    }

    (second == -21345678) ?
        printf("Second largest element not found ") :
        printf("second largest element is : %d", second);
    

    return 0;
}